### PR TITLE
SOLR-17482: Change long form of -r to --recursive

### PR DESCRIPTION
**Pull request by epugh**
Deprecate --recurse

https://issues.apache.org/jira/browse/SOLR-17482

# Description

See JIRA

# Solution

See JIRA

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [ ] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended, not available for branches on forks living under an organisation)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by epugh** [comment]
@malliaridis this is ready for review!

File: solr/core/src/test/org/apache/solr/cli/SolrCLIZkToolsTest.java
Comment (malliaridis) [inline solr/core/src/test/org/apache/solr/cli/SolrCLIZkToolsTest.java:1]: There are 2-3 more occurences in this file of the word "recurse" that should be replaced.

**Review by malliaridis — CHANGES_REQUESTED** [review]
Thanks for updating the documentation as well. 👍

There are a few more occurences I found:
- [ ] `ZkMaintenanceUtils#zkTransfer` (line 140) - `--recurse`
- [ ] `SolrCLIZkToolsTest` - any remaining occurence of the word "recurse" should be replaced (2-3 more still exist)

File: solr/core/src/test/org/apache/solr/cli/SolrCLIZkToolsTest.java
Comment (malliaridis) [inline solr/core/src/test/org/apache/solr/cli/SolrCLIZkToolsTest.java:1]: Line 175 is still using `--recurse`.

**Review by malliaridis — COMMENTED** [review]
One last occurence and we are good I think. :)

**Review by malliaridis — APPROVED** [review]

**Comment by malliaridis** [comment]
@epugh please note that `--recurse` may need adaptation for `branch_9x`. There are more references than in `main`.

**Pull request merged**